suite-common/trading/src/thunks/common/watchTradeThunk.ts

```diff
@@ -0,0 +1,137 @@
+import { createThunk } from '@suite-common/redux-utils';
+import { Account } from '@suite-common/wallet-types';
+
+import { TRADING_THUNK_PREFIX } from '../../constants';
+import { invityAPI } from '../../invityAPI';
+import { tradingActions } from '../../reducers/tradingReducer';
+import {
+    TradingTradeMapProps,
+    TradingTransaction,
+    TradingType,
+    TradingWatchTradeResponsePropsMap,
+} from '../../types';
+
+export type WatchTradeThunk = {
+    account: Account;
+    trade: TradingTransaction;
+    refreshCount: number;
+};
+
+type WatchTradeDataProps = {
+    trade: TradingTransaction;
+    refreshCount: number;
+};
+
+type WatchTradeDataResultProps<T extends TradingType> = {
+    tradeData: TradingTradeMapProps[T];
+    response: TradingWatchTradeResponsePropsMap[T];
+};
+
+const watchTradeData = async <T extends TradingType>({
+    trade,
+    refreshCount,
+}: WatchTradeDataProps): Promise<WatchTradeDataResultProps<T> | undefined> => {
+    const response = await invityAPI.watchTrade<T>(trade.data, trade.tradeType, refreshCount);
+
+    if (!response) return;
+
+    const isStatusSame = response.status && response.status === trade.data.status;
```

Comment:
The condition might be simplified to this, maybe?
```suggestion
if (response?.status === trade.data.status) {
	return
}
```